cmd/tink-worker/cmd/root.go

```diff
@@ -0,0 +1,171 @@
+			retries, _ := cmd.PersistentFlags().GetInt("retries")
+			// TODO(displague) is log-level no longer useful?
+			// logLevel, _ := cmd.PersistentFlags().GetString("log-level")
+			workerID, _ := cmd.PersistentFlags().GetString("id")
+			maxSize, _ := cmd.PersistentFlags().GetInt64("max-size")
+			timeOut, _ := cmd.PersistentFlags().GetDuration("timeout")
+			user, _ := cmd.PersistentFlags().GetString("registry-username")
+			pwd, _ := cmd.PersistentFlags().GetString("registry-password")
+			registry, _ := cmd.PersistentFlags().GetString("docker-registry")
+
+			logger.With("version", version).Info("starting")
+			if setupErr := client.Setup(); setupErr != nil {
+				return setupErr
+			}
+
+			ctx := context.Background()
+			if timeOut > 0 {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(ctx, timeOut)
+				defer cancel()
+			}
+
+			conn, err := tryClientConnection(logger, retryInterval, retries)
+			if err != nil {
+				return err
+			}
+			rClient := pb.NewWorkflowSvcClient(conn)
+
+			regConn := internal.NewRegistryConnDetails(registry, user, pwd, logger)
+			worker := internal.NewWorker(rClient, regConn, logger, registry, retries, retryInterval, maxSize)
+
+			err = worker.ProcessWorkflowActions(ctx, workerID)
+			if err != nil {
+				return errors.Wrap(err, "worker Finished with error")
+			}
+			return nil
+		},
+	}
+
+	rootCmd.PersistentFlags().Duration("retry-interval", retryIntervalDefault, "Retry interval in seconds")
+
+	rootCmd.PersistentFlags().Duration("timeout", 0, "Max duration to wait for worker to complete")
+
+	rootCmd.PersistentFlags().Int("max-retry", retryCountDefault, "Maximum number of retries to attempt")
+
+	rootCmd.PersistentFlags().Int64("max-file-size", defaultMaxFileSize, "Maximum file size in bytes")
+
+	// rootCmd.PersistentFlags().String("log-level", "info", "Sets the worker log level (panic, fatal, error, warn, info, debug, trace)")
+
+	rootCmd.PersistentFlags().StringP("id", "i", "", "Sets the worker id")
+	must(rootCmd.MarkPersistentFlagRequired("id"))
+
+	rootCmd.PersistentFlags().StringP("docker-registry", "r", "", "Sets the Docker registry")
+	must(rootCmd.MarkPersistentFlagRequired("docker-registry"))
+
+	rootCmd.PersistentFlags().StringP("registry-username", "u", "", "Sets the registry username")
+	must(rootCmd.MarkPersistentFlagRequired("registry-username"))
+
+	rootCmd.PersistentFlags().StringP("registry-password", "p", "", "Sets the registry-password")
+	must(rootCmd.MarkPersistentFlagRequired("registry-password"))
+
+	return rootCmd
+}
+
+// createViper creates a Viper object configured to read in configuration files
+// (from various paths with content type specific filename extensions) and load
+// environment variables that start with TINK_WORKER.
+func createViper() (*viper.Viper, error) {
+	v := viper.New()
+	v.AutomaticEnv()
+	v.SetConfigName("tink-worker")
+	v.AddConfigPath("/etc/tinkerbell")
+	v.AddConfigPath(".")
+	v.SetEnvPrefix("TINK_WORKER")
```

Comment:
mm, yes. I think I would argue in favor of `RETRY_INTERVAL`, `MAX_RETRY`, `DOCKER_REGISTRY`, `REGISTRY_USERNAME`, `REGISTRY_PASSWORD` having a `TINK_WORKER` prefix.

If I remove the prefix, `TINK_WORKER_ID` would become `ID` which is too ambiguous for env usage.

What does @tinkerbell/site-eng think?

Reply:
@mmlb ?

Reply:
I like having external things not have a prefix so that a common env can be setup and by used by all services (RETRY_INTERVAL, MAX_RETRY mostly). Kind of like being able to apply things globally or as wide as possible. Helps script things up too. Not strongly held though.

Reply:
It's certainly easier for this PR to not introduce breaking changes on the environment variables.

While I don't prefer the ambiguous names, I do prefer not breaking the API.

I'll update this to not include the `TINK_WORKER` prefix.